tls/s2n_client_hello.c

```diff
@@ -177,7 +177,7 @@ static int s2n_parse_client_hello(struct s2n_connection *conn)
 
     uint16_t cipher_suites_length = 0;
     GUARD(s2n_stuffer_read_uint16(in, &cipher_suites_length));
-    S2N_ERROR_IF(cipher_suites_length % S2N_TLS_CIPHER_SUITE_LEN, S2N_ERR_BAD_MESSAGE);
+    S2N_ERROR_IF(cipher_suites_length % S2N_TLS_CIPHER_SUITE_LEN || cipher_suites_length == 0, S2N_ERR_BAD_MESSAGE);
```

Comment:
I have a hard time reasoning about the nested checks in combination with the refutation logic here. Could we split it up and use `ENSURE_POSIX`:

```c
ENSURE_POSIX(cipher_suites_length > 0, S2N_ERR_BAD_MESSAGE);
ENSURE_POSIX(cipher_suites_length % S2N_TLS_CIPHER_SUITE_LEN == 0, S2N_ERR_BAD_MESSAGE);
```

Reply:
Yeah this is more readable